src/core/pool_type.hpp

```diff
@@ -136,6 +137,47 @@ struct Pool : PoolBase {
 		return ret;
 	}
 
+	/**
+	 * Iterator to iterate all valid T of a pool
+	 * @tparam T Type of the class/struct that is going to be iterated
+	 */
+	template <class T>
+	struct PoolIterator {
+		typedef T value_type;
+		typedef T* pointer;
+		typedef T& reference;
+		typedef size_t difference_type;
+		typedef std::forward_iterator_tag iterator_category;
```

Comment:
these all seem to be unused? Or are they necessary for the iterator "type" ?

Reply:
They are necessary if we want to use the iterator in std algorithms like std::distance() or std::count_if().
But they are not needed for range-based for loops.